compiler/src/iree/compiler/InputConversion/StableHLO/Passes.cpp

```diff
@@ -6,20 +6,122 @@
 
 #include "iree/compiler/InputConversion/StableHLO/Passes.h"
 
+#include "iree/compiler/Dialect/Util/Transforms/Passes.h"
+#include "iree/compiler/InputConversion/Common/Passes.h"
+#include "mlir/Conversion/ReconcileUnrealizedCasts/ReconcileUnrealizedCasts.h"
+#include "mlir/Conversion/SCFToControlFlow/SCFToControlFlow.h"
+#include "mlir/Conversion/ShapeToStandard/ShapeToStandard.h"
+#include "mlir/Dialect/SCF/Transforms/Passes.h"
+#include "mlir/Dialect/Shape/Transforms/Passes.h"
 #include "mlir/Pass/PassManager.h"
 #include "mlir/Pass/PassOptions.h"
 #include "mlir/Pass/PassRegistry.h"
+#include "mlir/Transforms/Passes.h"
 
 namespace mlir::iree_compiler::stablehlo {
-
 namespace {
 #define GEN_PASS_REGISTRATION
 #include "iree/compiler/InputConversion/StableHLO/Passes.h.inc"  // IWYU pragma: export
 }  // namespace
 
+namespace {
+// TODO(#8745): remove these flags when the -iree-flow-demote-* flags can be
+// used without tripping upstream verifier issues.
+llvm::cl::opt<bool> clDemoteI64ToI32(
+    "iree-stablehlo-demote-i64-to-i32",
+    llvm::cl::desc(
+        "Converts all StableHLO i64 ops and values into i32 counterparts."),
```

Comment:
I wonder if the situation with https://github.com/openxla/iree/issues/8745 has improved since it was filed...

Reply:
For now my goal is to reach feature-parity with the MHLO input conversion. I think this will be much easier if we do not try to optimize/fix too many thing along the way and risk having to undo it or debug potential differences.